react/Dialog/DialogEffects.ts

```diff
@@ -45,28 +46,37 @@ const makeOnUnmount = (
   immersive: boolean,
   sidebar: Element | null,
   cozyBar: Element | null
-): FlagshipUI => ({
-  ...(hasAnotherDialog
-    ? {}
-    : {
-        bottomBackground:
-          theme.palette.background[sidebar ? 'default' : 'paper']
-      }),
-  bottomTheme: immersive ? 'light' : 'dark',
-  bottomOverlay: 'transparent',
-  topOverlay: 'transparent',
-  ...(hasAnotherDialog || !cozyBar
-    ? {}
-    : {
-        topBackground:
-          cozyBar &&
-          getComputedStyle(cozyBar).getPropertyValue('background-color')
-      }),
-  topTheme:
-    immersive || (cozyBar && cozyBar.classList.contains('coz-theme-primary'))
+): FlagshipUI => {
+  const otherDialogStyles = getFirstDialogStyles()
+  const isDark =
+    otherDialogStyles !== null &&
+    lightOrDark(otherDialogStyles.color) === 'dark'
+
+  return {
+    ...(hasAnotherDialog
+      ? {}
+      : {
+          bottomBackground:
+            theme.palette.background[sidebar ? 'default' : 'paper']
+        }),
+    bottomTheme: isDark ? 'dark' : immersive ? 'light' : 'dark',
+    bottomOverlay: 'transparent',
+    topOverlay: 'transparent',
+    ...(hasAnotherDialog || !cozyBar
+      ? {}
+      : {
+          topBackground:
+            cozyBar &&
+            getComputedStyle(cozyBar).getPropertyValue('background-color')
+        }),
+    topTheme: isDark
+      ? 'dark'
```

Comment:
is it possible to simplify this line, to separate the dark case from the light scenario? 🙏

Reply:
rootModal is now injected in function param

Reply:
I was mentioning:

changing the current code looking like this: 

topTheme: condition : 'dark' : recondition ? 'light' : 'dark'
 
into

=> topTheme: condition? 'light': 'dark'

Reply:
Agreed